### PR TITLE
Precommit and Unit tests

### PR DESCRIPTION
**Pull request by fromflux**
Adds Jest dependencies and initial unit tests
Adds Precommit hook to run linting and unit tests

### CODE REVIEW

File: src/client/js/data/fetching.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  let mockedResponse = {
+    ok: mockedSuccess,
+    ...mockedPayload,
+    json: function() {
+      return mockedPayload
+    }
+  }
+
+  beforeEach(() => {
+    global.fetch = jest.fn().mockImplementation(() => {
```
Comment (fromflux) [inline src/client/js/data/fetching.test.js]: Is there a preferred fetch mocking method we can use?

File: src/client/js/data/fetching.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      expect(e.code).toEqual('HTTP_ERROR')
+    }
+  })
+
+  it('calls api with appropriate url', async () => {
+    mockedSuccess = true
+    mockedPayload = { foo: 'bar' }
+
+    await fetchData()
+    expect(global.fetch).toHaveBeenCalledWith('/api/stories?filter=undefined&')
```
Comment (fromflux) [inline src/client/js/data/fetching.test.js]: It does not look correct left to confirm

Comment (fromflux) [inline src/client/js/data/fetching.test.js]: Does not look correct but left to confirm

**Comment by ShogunPanda** [comment]
LGTM!

File: src/client/js/data/fetching.test.js (hunk 2 above)
Comment (chrisdwheatley) [reply src/client/js/data/fetching.test.js]: It looks like `fetchData` expects a filter to be passed in as the first argument, perhaps there needs to be more validation in the function, what do you think?

File: src/client/js/data/fetching.test.js (hunk 1 above)
Comment (fromflux) [reply src/client/js/data/fetching.test.js]: yeah, this is the only use of fetch on the client, so not sure if it is worth it

File: src/client/js/data/fetching.test.js (hunk 2 above)
Comment (fromflux) [reply src/client/js/data/fetching.test.js]: I'll rewrite for clarity, just wanted to confirm it was not intended

File: package.json [hunk 3]
```diff
@@ -52,18 +62,26 @@
...
   "devDependencies": {
     "@babel/cli": "^7.0.0-beta.46",
-    "@babel/core": "^7.0.0-beta.46",
+    "@babel/core": "^7.0.0-beta.49",
     "@babel/plugin-proposal-object-rest-spread": "^7.0.0-beta.46",
     "@babel/preset-env": "^7.0.0-beta.46",
     "@babel/preset-react": "^7.0.0-beta.46",
     "@babel/register": "^7.0.0-beta.46",
     "@cowtech/eslint-config": "^4.19.4",
+    "babel-core": "^7.0.0-bridge.0",
```
[the remark this answers is not in the document]
Comment (fromflux) [reply package.json]: they are likely the same (I'll try to confirm) package. the scoped version may be for beta/dev versions

File: package.json [hunk 4]
```diff
@@ -52,18 +62,26 @@
   },
   "devDependencies": {
     "@babel/cli": "^7.0.0-beta.46",
-    "@babel/core": "^7.0.0-beta.46",
+    "@babel/core": "^7.0.0-beta.49",
```
[the remark this answers is not in the document]
Comment (fromflux) [reply package.json]: not sure what you mean

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (fromflux) [reply package.json]: unfortunately we have dependencies on both names, we could try alias one into the other but I'm not sure we can do anything about it for now
https://github.com/babel/babel/blob/master/package.json

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisdwheatley) [reply package.json]: Update all the `@babel` scoped packages to `beta.49`.

[the remark this answers is not in the document]
Comment (fromflux) [reply package.json]: not sure that is safe in beta

**Pull request merged**